gh-113924: Add copy from zipfile without decompress/recompressing data (#125718)
python/cpython: Misc/NEWS.d/next/Library/2024-10-18-20-26-01.gh-issue-113924.ObB1IG.rst

> Add ``copy_file``, ``copy_files``, and ``copy_all_files`` to the ZipFile
> class. These methods allow copying from a zipfile without the overhead of
> decompressing and recompressing the data.

Reviewer: It's really too long, you should keep it short. Their functionality should be documented (you don't seem to have any documentation).
```suggestion
Added :meth:`.copy_file`, :meth:`.copy_files`, and :meth:`.copy_all_files` to the :class:`zipfile.ZipFile`. These methods allow copying from a zipfile.
```

Author: Thank you for reviewing my PR. I'll make the change you suggested here.

I planned on creating the documentation only after the API is approved by a core developer. This is my first time contributing to CPython, so I don't know if that's acceptable, however. Should I not wait to create the documentation?

Reviewer 2: There should be agreement that the feature is desired in the issue discussion before you work on a PR. Then, you know you can work on code, tests, documentation.

Also note that you don’t have to take reamrks from people who are not contributors to the project.
